src/api/createDeviceMessageHandlingSystem/src/v1/steps.py

```diff
@@ -26,6 +43,64 @@ def parse_mhs_device_payload(data, cache) -> Device:
     return CreateMhsDeviceIncomingParams(**payload)
 
 
+def get_party_key(data, cache) -> str:
+    product: CpmProduct = data[read_product]
+    party_keys = [
+        key.key_value
+        for key in product.keys
+        if key.key_type is ProductKeyType.PARTY_KEY
+    ]
+    try:
+        (party_key,) = party_keys
+    except ValueError:
+        raise NotEprProductError(
+            "Not an EPR Product: Cannot create MHS device for product without exactly one Party Key"
+        )
+    return party_key
+
+
+def check_for_existing_mhs(data, cache):
+    product_team: ProductTeam = data[read_product_team]
+    product: CpmProduct = data[read_product]
+
+    device_repo = DeviceRepository(
+        table_name=cache["DYNAMODB_TABLE"], dynamodb_client=cache["DYNAMODB_CLIENT"]
+    )
+    mhs_device = device_repo.search(
+        product_team_id=product_team.id, product_id=product.id
+    )
+    for device in mhs_device:
+        if device.name == "Product-MHS":
+            raise ConfigurationError(
+                f"There is already an exisitng MHS Device for this product, with device_id: {mhs_device[0].id}"
+            )
+
+
+def read_device_reference_data(data, cache) -> DeviceReferenceData:
+    path_params: CpmProductPathParams = data[parse_path_params]
+    drd_repo = DeviceReferenceDataRepository(
+        table_name=cache["DYNAMODB_TABLE"], dynamodb_client=cache["DYNAMODB_CLIENT"]
+    )
+    device_reference_data = drd_repo.search(
+        product_id=path_params.product_id,
+        product_team_id=path_params.product_team_id,
+    )
+
+    party_key: str = data[get_party_key]
+    mhs_message_set_drd_name = f"{party_key} - MHS Message Set"
+
+    device_reference_data = [
+        drd for drd in device_reference_data if drd.name == mhs_message_set_drd_name
+    ]
+
+    if len(device_reference_data) != 1:
```

Comment:
should this not be 1 or more? so 

`if not device_reference_data:`

This may be correct but I'm just going off what the first Scenario says

Reply:
Ok, maybe I'm wrong here. So it's one message set, but it can be multiple questionnaire responses?

Reply:
Yeah think the AC was just a general one, the actual task says

_if not exactly one Message Set DRD found then raise 5XX AccreditedSystemFatalError (shouldn't be possible)
returns the one Message Set DeviceReferenceData_

Because we only have one MHS per product, but you can have multiple AS's

Reply:
And again think there was some copy pasting overlap as the Error is MHS specific not Accredited System, and I also kept it as a 400 Validation error as that made more sense to me and would be more useful to the user I think